### PR TITLE
Help feature added & autosave system changes

### PR DESCRIPTION
**Pull request by GuzioMG**
I just changed some things in the autosave system on my private fork (now user can choose whenever to save on exit or no), and I decided to pull request to the original GitHub repository.
I also added help command, so you can list all commands now.
They are small changes in the command system, but I think, that it's a good idea.
By the way: small changes in README.MD file again.

### CODE REVIEW

File: NitroxServer/ConsoleCommands/ExitCommand.cs [hunk 1]
```diff
@@ -11,12 +13,66 @@ public ExitCommand() : base("exit", Optional<string>.Empty(), "Exits the server"
...
+            else
+            {
+                Console.WriteLine("Worng key. Calling \"Save\" method again...");
+                return Save();
+            }
+        }
+
+        bool AreYouSure()
+        {
+            Console.WriteLine("Are you sure? ALL UNSAVED PROGRESS WILL BE LOST! [Y/N]");
```
Comment (Measurity) [inline NitroxServer/ConsoleCommands/ExitCommand.cs]: I don't think it's a good idea to ask for console input in a command. These commands can be excuted in-game too and will break the server console. Instead I recommend just requiring a -y or -n argument to the command. If no arg is supplied you just tell the user to reenter the command with -y or -n.

File: NitroxServer/ConsoleCommands/ExitCommand.cs [hunk 2]
```diff
@@ -11,12 +13,66 @@ public ExitCommand() : base("exit", Optional<string>.Empty(), "Exits the server"
...
             return args.Length == 0;
         }
+        
+        bool Save()
+        {
+            Console.WriteLine("Save before exiting? [Y/N]");
+            char Output = Console.ReadKey(true).KeyChar;
+            if (Output == 'y')
+            {
+                Console.WriteLine("You pressed \"Y\" - Saving...");
```
Comment (Measurity) [inline NitroxServer/ConsoleCommands/ExitCommand.cs]: Use `Log` instead of console writeline

File: NitroxServer/ConsoleCommands/ExitCommand.cs [hunk 3]
```diff
@@ -11,12 +13,66 @@ public ExitCommand() : base("exit", Optional<string>.Empty(), "Exits the server"
 
         public override void RunCommand(string[] args)
         {
+            Console.WriteLine("Stopping server:");
```
Comment (Measurity) [inline NitroxServer/ConsoleCommands/ExitCommand.cs]: Use `Log` instead of `Console.WriteLine`.

**Comment by GuzioMG** [comment]
Thank you. I'll modify it.

**Comment by GuzioMG** [comment]
I'll upload update tomorrow, because it's 18:07 / 06:07 PM in Poland right now, and I have to learn for exam. I hope I'll finish editing tomorrow.

**Comment by Sunrunner37** [comment]
You'll also want to rebase these changes onto a new branch.  This PR is including all of your other commits that have already been merged.

**Comment by GuzioMG** [comment]
I also optimized help command a bit (now it gives itself instead of creating a new object), however, I didn't make command detection automatic yet, because I didn't have enough time to figure out how is the command processor actually working.

**Comment by Sunrunner37** [comment]
I'm going to close this PR.  Can you please re-open a new one with the fresh branch?

**Comment by GuzioMG** [comment]
Of course. I'll probably do that tomorrow.

**Comment by GuzioMG** [comment]
@Measurity You told, that I can ask you to help me If I have a problem. And I think, I have a problem. I created new branch (called new-master), and I made pull request, but old commits still displayed there. So I created next branch called "Problem" (because I have a problem). Here you have screenshot showing what's happening:
EDIT: I can't attach screenshot from phone, and I can't access to PC right now. But I think, you know, what I mean. Creating a new branch just don't makes any difference.

**Pull request closed without merging**